vortex-array/src/patches.rs

```diff
@@ -248,10 +355,63 @@ impl Patches {
                     .search_sorted(&needle, SearchSortedSide::Left);
             });
         }
-        self.indices.as_primitive_typed().search_sorted(
-            &PValue::U64((index + self.offset) as u64),
-            SearchSortedSide::Left,
-        )
+        indices
+            .as_primitive_typed()
+            .search_sorted(&PValue::U64(needle as u64), SearchSortedSide::Left)
+    }
+
+    /// Constant time searches for `index` in the indices array.
+    ///
+    /// First determines which chunk the target index falls into, then performs
+    /// a binary search within that chunk's range.
+    ///
+    /// Returns a [`SearchResult`] indicating either the exact patch index if found,
+    /// or the insertion point if not found.
+    ///
+    /// # Panics
+    /// Panics if `chunk_offsets` or `offset_within_chunk` are not set.
+    fn search_index_chunked(&self, index: usize) -> SearchResult {
+        let Some(chunk_offsets) = &self.chunk_offsets else {
+            vortex_panic!("chunk_offsets is required to be set")
+        };
+
+        let Some(offset_within_chunk) = self.offset_within_chunk else {
+            vortex_panic!("offset_within_chunk is required to be set")
+        };
+
+        let chunk_start_idx = (index + self.offset % PATCH_CHUNK_SIZE) / PATCH_CHUNK_SIZE;
```

Comment:
can you skip the mod here? I think the result of following is the same
```suggestion
        let chunk_start_idx = (index + self.offset) / PATCH_CHUNK_SIZE;
```

Reply:
No, `self.offset` is absolute and I want to find the offset within the chunk.

Reply:
I see, forgot that everything apart from offset is sliced